lib/src/modules/conditional.dart

```diff
@@ -1252,40 +1255,83 @@ class FlipFlop extends Module with CustomSystemVerilog {
   /// The clock, posedge triggered.
   late final Logic _clk = input(_clkName);
 
+  /// Optional enable input to the flop.
+  ///
+  /// If enable is  high or enable is not provided then flop output will vary
+  /// on the basis of clock [_clk] and input [_d]. If enable is low, then
+  /// output of the flop remains frozen irrespective of the input [_d].
+  late final Logic _en = input(_enName);
+
   /// The input to the flop.
   late final Logic _d = input(_dName);
 
   /// The output of the flop.
   late final Logic q = output(_qName);
 
+  /// To track if optional enable is provided or not.
+  late final bool _isEnableProvided;
+
   /// Constructs a flip flop which is positive edge triggered on [clk].
```

Comment:
We should discuss the impact of `en` in the doc comment of the constructor as well